### DISCUSSION TITLE
No notifications with Pushover

### DISCUSSION THREAD

**Discussion by cxhamilton** [containrrr/watchtower#753] [Q&A] [answered]
I am having trouble getting Pushover notifications from Watchtower. Reading the docs, there should be a notification on start-up and then additional notifications when images are updated. In my case, Watchtower is successfully starting, running, and updating images as expected but with no notifications.

Here is the environment for the container. I should mention that I am managing this and all other containers using Portainer, though I do not that makes a difference.

Key | Value
-- | --
TZ | America/New_York
WATCHTOWER_CLEANUP | true
NO_COLOR | true
WATCHTOWER_INCLUDE_STOPPED | true
WATCHTOWER_NOTIFICATIONS | shoutrrr
WATCHTOWER_NOTIFICATION_URL | pushover://shoutrrr:*APP_TOKEN*@*USER_TOKEN*
WATCHTOWER_DEBUG | true
PATH | /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
SHOUTRRR_DEBUG | true
WATCHTOWER_NOTIFICATIONS_LEVEL | trace
WATCHTOWER_TRACE | true

*I have omitted the APP_TOKEN and USER_TOKEN for obvious privacy reasons, but I have triple checked that they are correct (and they are 👍🏻)*

I have turned on as much logging as I know how and I also added SHOUTRRR_DEBUG=true after reading feedback from another post. Still, there are no notifications at start-up or when an image is updated.

Thoughts? Any advice would be greatly appreciated.

**Comment by piksel** [comment, marked as answer]
You need to provide at least one device to send to for pushover. It *should* send to all devices if none are specified, but that was probably missed during development.

**Reply by piksel** [reply]
Created an issue for shoutrrr, since this isn't particularly apparent for the end user
https://github.com/containrrr/shoutrrr/issues/111

**Reply by cxhamilton** [reply]
@piksel I have made the change you suggested. Should I see a Pushover notification on start-up? Unfortunately, I have not had an image that needs/requires updating (yet).

Key | Value
--|--
WATCHTOWER_NOTIFICATION_URL | pushover://shoutrrr:*API_TOKEN*@*USER_TOKEN*/?devices=iPhone

**Reply by piksel** [reply]
Yeah, I think it should show one on startup, this is the last one I got before my pushover subscription ended:
![image](https://user-images.githubusercontent.com/807383/104225612-e9176300-5446-11eb-85ed-6e9cb2912d9b.png)

What does the container log say? You can see them by using
```
 docker logs watchtower
 ```

**Reply by cxhamilton** [reply]
```
$ sudo docker logs watchtower
time="2021-01-11T15:43:43-05:00" level=debug
time="2021-01-11T15:43:43-05:00" level=debug msg="Sleeping for a second to ensure the docker api client has been properly initialized."
time="2021-01-11T15:43:44-05:00" level=debug msg="Retrieving running, stopped and exited containers"
time="2021-01-11T15:43:44-05:00" level=debug msg="There are no additional watchtower containers"
time="2021-01-11T15:43:44-05:00" level=debug msg="Watchtower HTTP API skipped."
time="2021-01-11T15:43:44-05:00" level=info msg="Starting Watchtower and scheduling first run: 2021-01-11 18:00:00 -0500 EST"
```

Key | Value
-- | --
WATCHTOWER_DEBUG | true
SHOUTRRR_DEBUG | true
WATCHTOWER_NOTIFICATIONS_LEVEL | trace
WATCHTOWER_TRACE | true

**Reply by piksel** [reply]
Yep, was able to reproduce with
`WATCHTOWER_NOTIFICATIONS_LEVEL="trace"`

Try running without it (none of those you just posted should be necessary, except for the `WATCHTOWER_DEBUG` when you want more details in your logs).

I will continue to look into why that fully turns off notifications, but you probably don't want that level anyway. It gets real spammy.
